src/MongoDB/ReadPreference.c

```diff
@@ -74,22 +74,39 @@ PHP_METHOD(ReadPreference, __construct)
 		case MONGOC_READ_SECONDARY_PREFERRED:
 		case MONGOC_READ_NEAREST:
 			intern->read_preference = mongoc_read_prefs_new(mode);
+			break;
+		default:
+			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Invalid mode: %" PHONGO_LONG_FORMAT, mode);
+			return;
+	}
 
+	switch(ZEND_NUM_ARGS()) {
```

Comment:
This single-case `switch` should make sense once you see https://github.com/mongodb/mongo-php-driver/pull/400. Since I was refactoring the tagSet validation already, I opted to restructure this early. It should also make the diff for PHPC-752's commit easier to understand.